common/remote/sockfile.cpp

```diff
@@ -2697,13 +2697,26 @@ ISocket *checkSocketSecure(ISocket *socket)
 
     if ( (pport == securitySettings.daFileSrvSSLPort) && (!socket->isSecure()) )
     {
-        Owned<ISecureSocket> ssock;
 #ifdef _USE_OPENSSL
-        ssock.setown(createSecureSocket(LINK(socket), ClientSocket));
-        int status = ssock->secure_connect();
-        if (status < 0)
-            throw createDafsException(DAFSERR_connection_failed,"Failure to establish secure connection");
-        return ssock.getClear();
+        Owned<ISecureSocket> ssock;
+        try
+        {
+            ssock.setown(createSecureSocket(LINK(socket), ClientSocket));
+            int status = ssock->secure_connect();
+            if (status < 0)
+                throw MakeStringException(-1, "Failure to establish secure connection");
```

Comment:
Ok, it makes no difference really.  The catch below catches both ways, just curious about why it changed.  I do think you should avoid using -1 as the code.  You will find lots of those around, but mostly out of past lazyness.  We should try to use/define error codes for all exceptions.  DAFSERR_connection_failed should be fine even if you don't use createDafsException here.